src/pointer.cr

```diff
@@ -242,7 +242,7 @@ struct Pointer(T)
     raise ArgumentError.new("Negative count") if count < 0
 
     if self.class == source.class
-      Intrinsics.memcpy(self.as(Void*), source.as(Void*), (count * sizeof(T)).to_u32, 0_u32, false)
+      Intrinsics.memcpy(self.as(Void*), source.as(Void*), Intrinsics::SizeT.new(count) * sizeof(T), 0_u32, false)
```

Comment:
Why introduce `Intrinsics::SizeT` but use `LibC::SizeT` below? This is confusing *and* a duplication.